uvm/impl/com/untangle/uvm/AppManagerImpl.java

```diff
@@ -1105,13 +1107,18 @@ public void run()
 
                     if (!UvmContextFactory.context().licenseManager().isLicenseValid(name)) {
 
-                        logger.info("Stopping  : " + name + " [" + id + "] because of invalid license");
+                        logger.info("Stopping : " + name + " [" + id + "] because of invalid license");
 
                         long startTime = System.currentTimeMillis();
                         ((AppBase) app).stopIfRunning();
                         long endTime = System.currentTimeMillis();
 
-                        logger.info("Stopped   : " + name + " [" + id + "] [" + (((float) (endTime - startTime)) / 1000.0f) + " seconds]");
+                        logger.info("Stopped : " + name + " [" + id + "] [" + (((float) (endTime - startTime)) / 1000.0f) + " seconds]");
+                    } else {
+                        if (app.getRunState() == AppSettings.AppState.INITIALIZED) {
```

Comment:
If the user had stopped the app but didn't uninstall it, will this logic restart the app when the licenses are refreshed?

Reply:
The use case is to not install anything without internet, but that can change if you think it'd be better to have apps installed but not valid.